runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -208,33 +212,67 @@ jvmtiGetThreadListStackTraces(jvmtiEnv* env,
 		ENSURE_NON_NEGATIVE(max_frame_count);
 		ENSURE_NON_NULL(stack_info_ptr);
 
+#if JAVA_SPEC_VERSION >= 19
+		while (0 != thread_count) {
+			jthread thread = *thread_list;
+
+			if ((NULL != thread) && IS_VIRTUAL_THREAD(currentThread, J9_JNI_UNWRAP_REFERENCE(thread))) {
+				J9VMThread *targetThread = NULL;
+				getVMThread(currentThread, thread, &targetThread, TRUE, FALSE);
+			}
+			++thread_list;
+			--thread_count;
+		}
+#endif /* JAVA_SPEC_VERSION >= 19 */
+
 		vm->internalVMFunctions->acquireExclusiveVMAccess(currentThread);
 
+#if JAVA_SPEC_VERSION >= 19
+		thread_count = originalThreadCount;
+		thread_list = originalThreadList;
+#endif /* JAVA_SPEC_VERSION >= 19 */
+
 		stackInfo = j9mem_allocate_memory(((sizeof(jvmtiStackInfo) + (max_frame_count * sizeof(jvmtiFrameInfo))) * thread_count) + sizeof(jlocation), J9MEM_CATEGORY_JVMTI_ALLOCATE);
 		if (stackInfo == NULL) {
 			rc = JVMTI_ERROR_OUT_OF_MEMORY;
 		} else {
-			jvmtiFrameInfo * currentFrameInfo = (jvmtiFrameInfo *) ((((UDATA) (stackInfo + thread_count)) + sizeof(jlocation)) & ~sizeof(jlocation));
-			jvmtiStackInfo * currentStackInfo = stackInfo;
+			jvmtiFrameInfo *currentFrameInfo = (jvmtiFrameInfo *)((((UDATA)(stackInfo + thread_count)) + sizeof(jlocation)) & ~sizeof(jlocation));
+			jvmtiStackInfo *currentStackInfo = stackInfo;
 
-			while (thread_count != 0) {
+			while (0 != thread_count) {
 				jthread thread = *thread_list;
-				J9VMThread * targetThread;
-				j9object_t threadObject;
+				J9VMThread *targetThread = NULL;
+				j9object_t threadObject = NULL;
+				BOOLEAN isThreadAlive = FALSE;
 
-				if (thread == NULL) {
+				if (NULL == thread) {
 					rc = JVMTI_ERROR_NULL_POINTER;
 					goto deallocate;
 				}
 
-				if (!isSameOrSuperClassOf(J9VMJAVALANGTHREAD_OR_NULL(vm), J9OBJECT_CLAZZ(currentThread, *((j9object_t *) thread)))) {
+				if (!isSameOrSuperClassOf(J9VMJAVALANGTHREAD_OR_NULL(vm), J9OBJECT_CLAZZ(currentThread, *((j9object_t *)thread)))) {
 					rc = JVMTI_ERROR_INVALID_THREAD;
 					goto deallocate;
 				}
 
-				threadObject = *((j9object_t*) thread);
+				threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
 				targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, threadObject);
```

Comment:
Can you move this line inside the if check's else case (Line 272) before the NULL check